content/document.js

```diff
@@ -206,16 +193,42 @@ function update(contentRoot, folder, rawHtml, metadata) {
   }
 }
 
-function del(contentRoot, folder) {
-  const { metadata } = read(contentRoot, folder);
+function del(folder) {
+  const { metadata } = read(folder, { metadata: true });
   fs.rmdirSync(folder, { recursive: true });
-  updateWikiHistory(path.join(contentRoot, metadata.locale), metadata.slug);
+  updateWikiHistory(path.join(CONTENT_ROOT, metadata.locale), metadata.slug);
+}
+
+const findByURL = memoizeDuringBuild((url, fields = null) => {
+  const folder = urlToFolderPath(url);
+
+  const document = read(folder, fields);
+
+  return document ? { contentRoot: CONTENT_ROOT, folder, document } : null;
+});
+
+function findChildren(url, fields = null) {
+  const folder = urlToFolderPath(url);
+  const childPaths = glob.sync(
+    path.join(CONTENT_ROOT, folder, "*", HTML_FILENAME),
+    {
+      ignore: path.join(CONTENT_ROOT, getHTMLPath(folder)),
+    }
```

Comment:
I don't think this is needed, at least it worked for me without it.

Reply:
Indeed!